Formula/libsolv.rb

```diff
@@ -0,0 +1,60 @@
+class Libsolv < Formula
+  desc "Library for solving packages and reading repositories"
+  homepage "https://github.com/openSUSE/libsolv"
+  url "https://github.com/openSUSE/libsolv/archive/refs/tags/0.7.23.tar.gz"
+  sha256 "0286155964373c6cc3802d025750786c3ee79608d5cb884598e110e3918bb2fe"
+  license "BSD-3-Clause"
+
+  depends_on "cmake" => :build
+  depends_on "xz"
+  depends_on "zstd"
+
+  uses_from_macos "bzip2"
+  uses_from_macos "expat"
+  uses_from_macos "zlib"
+
+  def install
+    args = %w[
+      -DENABLE_STATIC=ON
+      -DENABLE_SUSEREPO=ON
+      -DENABLE_COMPS=ON
+      -DENABLE_HELIXREPO=ON
+      -DENABLE_DEBIAN=ON
+      -DENABLE_MDKREPO=ON
+      -DENABLE_ARCHREPO=ON
+      -DENABLE_CUDFREPO=ON
+      -DENABLE_CONDA=ON
+      -DENABLE_APPDATA=ON
+      -DMULTI_SEMANTICS=ON
+      -DENABLE_LZMA_COMPRESSION=ON
+      -DENABLE_BZIP2_COMPRESSION=ON
+      -DENABLE_ZSTD_COMPRESSION=ON
+      -DENABLE_ZCHUNK_COMPRESSION=ON
+    ]
+
+    args << "-DCMAKE_INSTALL_RPATH=#{rpath}"
+
+    mkdir "build" do
+      system "cmake", "..", *args, *std_cmake_args
+      system "make"
+      system "make", "install"
+    end
```

Comment:
```suggestion
    system "cmake", "-S", ".", "-B", "build", *args, *std_cmake_args
    system "cmake", "--build", "build"
    system "cmake", "--install", "build"
```
Let's follow the template from `brew create --cmake`.